pkg/jsonSchemaValidator/extensions/customKeyRegoDefinition.go

```diff
@@ -0,0 +1,116 @@
+// This file defines a custom key to implement the logic for rego rule:
+
+package jsonSchemaValidator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+const RegoDefinitionCustomKey = "regoDefinition"
+
+var regoCodeToEval = RegoDefinition{}
```

Comment:
im not sure about the global var that all function can modify it

same case but a lil bit different, maybe it can explain better: 
https://gobyexample.com/mutexes

Reply:
Basically I need some way to preserve the rego code I get in the Compile function. Basically only 1 place in the code sets and no one updated it. What is the concern?